### ISSUE TITLE
Detect missing associated types in trait impls

### ISSUE THREAD

**Issue by alygin** [intellij-rust/intellij-rust#1010]
There's the "Missing trait item" annotation that detects which functions are missing in a trait implementation. It would be nice to also detect missing associated types:

```rust
struct Foo;
trait Bar {
    type SomeType;
}
impl Bar for Foo {  // <-- should complain about missing `SomeType`
}
```

Even better would be to add the missing associated types in the fix, provided by the existing annotation.

**Comment by alygin** [comment]
closed in https://github.com/intellij-rust/intellij-rust/commit/75a77e2b5027be8f7cb54aa4c90379c1943cc35c
